frontend/src/components/CustomTimePicker/timezoneUtils.ts

```diff
@@ -0,0 +1,139 @@
+	BROWSER: 'BROWSER',
+	UTC: 'UTC',
+	STANDARD: 'STANDARD',
+} as const;
+
+type TimezoneType = typeof TIMEZONE_TYPES[keyof typeof TIMEZONE_TYPES];
+
+const UTC_TIMEZONE: Timezone = {
+	name: 'Coordinated Universal Time — UTC, GMT',
+	value: 'UTC',
+	offset: 'UTC',
+	searchIndex: 'UTC',
+	hasDivider: true,
+};
+
+// Helper functions
+const isValidTimezone = (tzName: string): boolean => {
+	try {
+		dayjs.tz(dayjs(), tzName);
+		return true;
+	} catch {
+		return false;
+	}
+};
+
+const formatOffset = (offsetMinutes: number): string => {
+	if (offsetMinutes === 0) return 'UTC';
+
+	const hours = Math.floor(Math.abs(offsetMinutes) / 60);
+	const minutes = Math.abs(offsetMinutes) % 60;
+	const sign = offsetMinutes > 0 ? '+' : '-';
+
+	return `UTC ${sign} ${hours}${
+		minutes ? `:${minutes.toString().padStart(2, '0')}` : ':00'
+	}`;
+};
+
+const createTimezoneEntry = (
+	name: string,
+	offsetMinutes: number,
+	type: TimezoneType = TIMEZONE_TYPES.STANDARD,
+	hasDivider = false,
+): Timezone => {
+	const offset = formatOffset(offsetMinutes);
+	let value = name;
+	let displayName = name;
+
+	switch (type) {
+		case TIMEZONE_TYPES.BROWSER:
+			displayName = `Browser time — ${name}`;
+			value = name;
+			break;
+		case TIMEZONE_TYPES.UTC:
+			displayName = 'Coordinated Universal Time — UTC, GMT';
+			value = 'UTC';
+			break;
+		case TIMEZONE_TYPES.STANDARD:
+			displayName = name;
+			value = name;
+			break;
+		default:
+			console.error(`Invalid timezone type: ${type}`);
+	}
+
+	return {
+		name: displayName,
+		value,
+		offset,
+		searchIndex: offset.replace(/ /g, ''),
+		...(hasDivider && { hasDivider }),
+	};
+};
+
+const getOffsetByTimezone = (timezone: string): number => {
+	const dayjsTimezone = dayjs().tz(timezone);
+	return dayjsTimezone.utcOffset();
+};
+
+const filterAndSortTimezones = (
+	allTimezones: ReturnType<typeof getTimeZones>,
+	browserTzName?: string,
+): Timezone[] =>
+	allTimezones
+		.filter(
+			(tz) =>
+				!tz.name.startsWith('Etc/') &&
+				isValidTimezone(tz.name) &&
+				tz.name !== browserTzName,
+		)
+		.sort((a, b) => a.name.localeCompare(b.name))
+		.map((tz) => createTimezoneEntry(tz.name, tz.rawOffsetInMinutes));
+
+const generateTimezoneData = (): Timezone[] => {
+	const allTimezones = getTimeZones();
+	const timezones: Timezone[] = [];
+
+	// Add browser timezone
+	const browserTz = dayjs.tz.guess();
+	const browserOffset = getOffsetByTimezone(browserTz);
+	timezones.push(
+		createTimezoneEntry(browserTz, browserOffset, TIMEZONE_TYPES.BROWSER),
+	);
+
+	// Add UTC timezone with divider
+	timezones.push(UTC_TIMEZONE);
+
+	// Add remaining timezones
+	timezones.push(...filterAndSortTimezones(allTimezones, browserTz));
+
+	return timezones;
+};
+
+export const getTimezoneObjectByTimezoneString = (
+	timezone: string,
+): Timezone | null => {
+	const utcOffset = getOffsetByTimezone(timezone);
+
+	return createTimezoneEntry(timezone, utcOffset);
+};
+
+export const TIMEZONE_DATA = generateTimezoneData();
```

Comment:
I don't like the idea of this being seen and used as a constant - `TIMEZONE_DATA`, but we are recomputing the timezone list every time.

Reply:
the `TIMEZONE_DATA` is not going to change, and it won't be recomputed (it will be computed only when the module is first imported). I guess we can keep the `TIMEZONE_DATA`.

Reply:
https://github.com/user-attachments/assets/d0f45985-bd51-42b4-8251-87baf0024b57